sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/IntervalExpressionsSuite.scala

```diff
@@ -274,4 +276,26 @@ class IntervalExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
       checkException(weeks = Int.MaxValue)
     }
   }
+
+  test("SPARK-34824: multiply year-month interval by numeric") {
```

Comment:
Should we also add `checkConsistencyBetweenInterpretedAndCodegen`

Reply:
@tanelk Thank you for review. I will add the check.